### ISSUE TITLE
Unable to use hooks in view components

### ISSUE THREAD

**Issue by tiberiuichim** [elastic/search-ui#588]
The way the components are rendered here is wrong:

https://github.com/elastic/search-ui/blob/6a7cfffa65e0cddadf2fe1751f5a75ff3bff2096/packages/react-search-ui/src/containers/Facet.js#L110

It shouldn't be called as a function, it should be called as `<View ...>`. 

Several drawbacks of the current code:

- the components are unmounted on refresh
- unable to use hooks
- I have to wrap these view components in another component, just so that I can use hooks.

**Comment by JasonStoltz** [comment]
You're absolutely right. This needs to be fixed. I've been strapped with other work. I'd welcome a PR to get this moving if anyone has time.
